src/Workspaces/Remote/ServiceHub/Shared/Extensions.cs

```diff
@@ -138,25 +138,27 @@ internal static partial class Extensions
 
                     return result;
                 }
-            }
-            catch (Exception ex) when (IsCancelled(ex))
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-                throw;
+                catch (Exception ex) when (IsCancelled(ex, mergedCancellation.Token))
```

Comment:
ya, it is linked cancellationtoken. so merged one will be raised when cancellation token is raised.